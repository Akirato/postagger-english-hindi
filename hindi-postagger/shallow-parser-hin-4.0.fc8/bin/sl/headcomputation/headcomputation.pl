#!/usr/bin/perl

use Getopt::Long;
GetOptions("help!"=>\$help,"path=s"=>\$head_home,"input=s"=>\$input,"output=s",\$output);
print "Unprocessed by Getopt::Long\n" if $ARGV[0];
foreach (@ARGV) {
	print "$_\n";
	exit(0);
}

if($help eq 1)
{
	print "Head Computation  - Head Computation Version 1.8\n(30th May 2009)\n\n";
	print "usage : ./run-headCompute.pl --path=/home/headComputation-1.8 [-i inputfile|--input=\"input_file\"] [-o outputfile|--output=\"output_file\"] \n";
	print "\tIf the output file is not mentioned then the output will be printed to STDOUT\n";
	exit(0);
}

if($head_home eq "")
{
	print "Please Specify the Path as defined in --help\n";
	exit(0);

}


my $src=$head_home . "/src";
require "$head_home/API/shakti_tree_api.pl";
require "$head_home/API/feature_filter.pl";
require "$src/copy_np_head.pl";
require "$src/copy_vg_head.pl";
require "$src/single_quote_changeName-0.1.pl";


if ($input eq "")
{
  $input="/dev/stdin";
}


&read_story($input);

$numBody = &get_bodycount();
for(my($bodyNum)=1;$bodyNum<=$numBody;$bodyNum++)
{

	$body = &get_body($bodyNum,$body);

# Count the number of Paragraphs in the story
	my($numPara) = &get_paracount($body);

#print STDERR "Paras : $numPara\n";

# Iterate through paragraphs in the story
	for(my($i)=1;$i<=$numPara;$i++)
	{

		my($para);
		# Read Paragraph
		$para = &get_para($i);


		# Count the number of sentences in this paragraph
		my($numSent) = &get_sentcount($para);
	#	print STDERR "\n $i no.of sent $numSent";

		#print STDERR "Para Number $i, Num Sentences $numSent\n";

		#print $numSent."\n";

		# Iterate through sentences in the paragraph
		for(my($j)=1;$j<=$numSent;$j++)
		{

			#print " ... Processing sent $j\n";

			# Read the sentence which is in SSF format
			my($sent) = &get_sent($para,$j);
			#print STDERR "$sent";
		#	print "check--\n";
		#	&print_tree($sent);
			# Get the nodes of the sentence (words in our case)


			#Copy NP head
		#	&AddID($sent);
			&make_chunk_name($sent);
			&copy_np_head($sent,$head_home);
			#Copy NP VG head
			&copy_vg_head($sent,$head_home);

		}
	}
}

if($output eq "")
{
	&printstory();
}

if($output ne "")
{
	&printstory_file("$output");
}

