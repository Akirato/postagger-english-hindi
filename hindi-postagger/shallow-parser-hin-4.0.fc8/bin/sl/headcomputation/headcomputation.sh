perl $SHALLOW_PARSER_HIN/bin/sl/headcomputation/printinput.pl $1 > /tmp/headcomputationinput$$.tmp
#perl $setu/bin/sys/common/printinput.pl $1 > headcomputationinput
#perl $setu/bin/sl/headcomputation/headcomputation.pl --path=$setu/bin/sl/headcomputation --input=headcomputationinput
perl $SHALLOW_PARSER_HIN/bin/sl/headcomputation/headcomputation.pl --path=$SHALLOW_PARSER_HIN/bin/sl/headcomputation --input=/tmp/headcomputationinput$$.tmp
rm -fr /tmp/headcomputationinput$$.tmp
