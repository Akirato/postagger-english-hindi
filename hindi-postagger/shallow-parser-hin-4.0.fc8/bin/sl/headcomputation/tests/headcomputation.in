<Sentence id="1">
1	((	NP	
1.1	BArawIya	JJ	<fs af='BArawIya,n,m,sg,,o,,'>
1.2	saMskqwi	NN	<fs af='saMskqwi,n,f,sg,,o,,'>
1.3	meM	PSP	<fs af='meM,pn,,,,,,'>
	))		
2	((	NP	
2.1	parvoM	NN	<fs af='parva,n,m,pl,,d,,'>
2.2	kA	PSP	<fs af='kA,n,m,sg,,o,7,'>
	))		
3	((	NP	
3.1	viSeRa	JJ	<fs af='viSeRa,n,m,sg,,o,,'>
3.2	sWAna	NN	<fs af='sWAna,n,m,sg,,o,,'>
	))		
4	((	VGF	
4.1	hE	VAUX	<fs af='hE,v,any,sg,2,,,hE'>
4.2	.	SYM	<fs af='&sdot;,punc,,,,,,'>
	))		
</Sentence>
