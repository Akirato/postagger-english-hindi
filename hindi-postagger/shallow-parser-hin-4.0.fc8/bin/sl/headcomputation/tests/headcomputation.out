<Sentence id="1">
1	((	NP	<fs af='saMskqwi,n,f,sg,,o,,' head=saMskqwi>
1.1	BArawIya	JJ	<fs af='BArawIya,n,m,sg,,o,,' name=BArawIya>
1.2	saMskqwi	NN	<fs af='saMskqwi,n,f,sg,,o,,' name=saMskqwi>
1.3	meM	PSP	<fs af='meM,pn,,,,,,' name=meM>
	))		
2	((	NP	<fs af='parva,n,m,pl,,d,,' head=parvoM>
2.1	parvoM	NN	<fs af='parva,n,m,pl,,d,,' name=parvoM>
2.2	kA	PSP	<fs af='kA,n,m,sg,,o,7,' name=kA>
	))		
3	((	NP	<fs af='sWAna,n,m,sg,,o,,' head=sWAna>
3.1	viSeRa	JJ	<fs af='viSeRa,n,m,sg,,o,,' name=viSeRa>
3.2	sWAna	NN	<fs af='sWAna,n,m,sg,,o,,' name=sWAna>
	))		
4	((	VGF	<fs af='hE,v,any,sg,2,,,hE' head=hE>
4.1	hE	VAUX	<fs af='hE,v,any,sg,2,,,hE' name=hE>
4.2	.	SYM	<fs af='&sdot;,punc,,,,,,' name=.>
	))		
</Sentence>
